test/services_manager_target_test.rb

```diff
@@ -100,34 +101,51 @@ class Template < Struct.new(
         subject.default_target = target
       end
 
-      context "when the default target has not been changed" do
-        let(:target) { "graphical" }
+      context "default target is available" do
+        before do
+          allow(Yast2::Systemd::Target).to receive(:find).and_return(TestTarget::GRAPHICAL)
+        end
 
-        it "does not perform changes in the underlying system" do
-          expect(Yast2::Systemd::Target).to_not receive(:set_default)
+        context "when the default target has not been changed" do
+          let(:target) { "multi-user" }
 
-          subject.save
+          it "does not perform changes in the underlying system" do
+            expect(Yast2::Systemd::Target).to_not receive(:set_default)
+
+            subject.save
+          end
+        end
+
+        context "when the default target has been changed" do
+          let(:target) { "graphical" }
+
+          it "saves the changes in the underlying system" do
+            expect(Yast2::Systemd::Target).to receive(:set_default).with("graphical")
+
+            subject.save
+          end
         end
       end
 
-      context "when the default target has been changed" do
-        let(:target) { "multi-user" }
+      context "when default target is not available" do
+        let(:target) { "graphical" }
 
-        it "saves the changes in the underlying system" do
+        it "reports a warning and sets to multi-user" do
+          expect(Yast2::Systemd::Target).to receive(:find).and_return(nil)
```

Comment:
I guess you are using this line to make the target unavailable. If so, I prefer to do it in a before block, that is:

```suggestion

context "when default target is not available" do
  let(:target) { "graphical" }

  before do
    allow(Yast2::Systemd::Target).to receive(:find).with(target).and_return(nil)
  end

  it "reports a warning" do
    expect(Yast::Report).to receive(:Warning)
    
    subject.save
  end

  it "sets 'multi-user' as default target" do
    expect(Yast2::Systemd::Target).to receive(:set_default).with("multi-user")

    subject.save
  end
end

````

Reply:
That does not work because you are calling subject.save twice and after the first call you have already set the default_target to mutli-user. I must admit, my first try has also looked like that :-)

Reply:
Ah, I see. If this happens again in the future, maybe a solution is to reset the default_target inside the `before` block.